agents/getters/getCurrentAgentGroupMemberProfiles.js

```diff
@@ -0,0 +1,19 @@
+import { createSelector } from 'reselect'
+import { filter, map, prop, values, contains } from 'ramda'
+
+import getProfiles from 'dogstack-agents/profiles/getters/getProfiles'
+import getCurrentAgentGroupMemberIds from './getCurrentAgentGroupMemberIds'
+
+const getCurrentAgentGroupMemberProfiles = createSelector(
```

Comment:
hmmm - just checking in on this @ahdinosaur - as far as I can see, we will still need all these getters, otherwise the `shouldQueryAgain` function won't necessarily know whether to query the db or not?
i.e. a getter like `getCurrentAgentGroupSupplierProfiles` is used in that function to make sure we have the profiles of the suppliers of the groups of the currentAgent?
or maybe i'm not understanding what you mean by 'instead creating getters which transforms agent id's from corresponding getters into agents from getAgents'...

Reply:
@iainkirkpatrick i'm not saying get rid of these getters completely, i'm saying i recommend having getters that use the `getAgents` getter to return _all the data for the same agents ids_. so instead of `getCurrentAgentGroupMemberProfiles`, it'd be `getCurrentAgentGroupMembers`, where each value is the full agent object with profile, credential, relationships, etc. does that make more sense?

Reply:
something like

```js

const getCurrentAgentGroupMemberProfiles = createSelector(
  getAgents,
  getCurrentAgentGroupMemberIds,
  (agents, memberIds) => {
    return memberIds.map(memberId => agents[memberId])
  }
)
```

Reply:
@ahdinosaur it does, but i'm still not sure how then we can manage the query loop - if we don't have some kind of check in `shouldQueryAgain` for whether the profiles of those members have loaded from the database, then the right condition in the `query` function won't ever be triggered to go and load the profiles?

maybe what i'm seeing here is that we are using getters for 2 purposes in `feathers-action-react` - for yanking data from state to give to wrapped components, but also to know whether we have the right data loaded yet to either stop querying or to activate new queries. I can see the point in not having `getCurrentAgentGroupMemberProfiles` for the first purpose, but don't see how we can avoid having it for the second? (could write conditional logic, but it'd be basically the same thing as the getter anyway?)

Reply:
it's some pattern like state -> state by relation -> enhanced data from composed state -> enhanced data by relation -> use case specific data

Reply:
done :)